File: 153_find_minimum_rotated_sorted_array/find_minimum_rotated_sorted_array.go
package main

import "fmt"

type Solution struct{}

func (s *Solution) findMinimumRotatedSortedArray(nums []int) int {
	left, right := 0, len(nums)-1

	for left < right {
		mid := (left + right) / 2
		if nums[mid] > nums[right] {
			left = mid + 1
		} else {
			right = mid
		}
	}
	return nums[left]
}

func main() {
	nums := []int{4, 5, 6, 7, 0, 1, 2}
	s := Solution{}
	fmt.Println(s.findMinimumRotatedSortedArray(nums))
}

File: 153_find_minimum_rotated_sorted_array/find_minimum_rotated_sorted_array.py
from typing import List

class Solution:
    def find_minimum_rotated_sorted_array(self, nums: List[int]) -> int:
        left, right = 0, len(nums) - 1

        while left < right:
            mid = (left + right) // 2
            if nums[mid] > nums[right]:
                left = mid + 1
            else:
                right = mid
        return nums[left]


if __name__ == "__main__":
    nums = [4, 5, 6, 7, 0, 1, 2]
    solution = Solution()
    print(solution.find_minimum_rotated_sorted_array(nums))
